src/models/Thought.js

```diff
@@ -0,0 +1,37 @@
+const {Schema, model} = require("mongoose");
+
+const moment = require("moment");
+
+const reactions = require("./Reaction");
+
+const thoughtSchema = {
+	thoughtText: {
+		type: String,
+		required: true,
+		minLength: 1,
+		maxLength: 280,
+	},
+	createdAt: {
+		type: Date,
+		default: moment(),
+	},
+	username: {
+		type: String,
+		required: true,
+	},
+	reactions: [reactions],
+};
+
+const schema = new Schema(thoughtSchema, {
+	toJSON: {
+		getters: true,
+	},
+});
+
+schema.virtual("reactionCount").get(function () {
+	return this.reactions.length;
+});
+
+const Thought = model("thought", schema);
```

Comment:
remember that your model names should be capitalised everywhere - `const Thought = model("Thought", schema);`